mobile/src/main/java/org/openhab/habdroid/ui/ItemPickerActivity.java

```diff
@@ -0,0 +1,272 @@
+        Log.d(TAG, "onResume()");
+        loadItems();
+    }
+
+    @Override
+    public void onPause() {
+        super.onPause();
+        Log.d(TAG, "onPause()");
+        // Cancel request for items if there was any
+        if (mRequestHandle != null) {
+            mRequestHandle.cancel();
+        }
+    }
+
+    public boolean onOptionsItemSelected(MenuItem item) {
+        Log.d(TAG, "onOptionsItemSelected()");
+        if (item.getItemId() == android.R.id.home) {
+            finish(false, null, null);
+            return true;
+        }
+        return super.onOptionsItemSelected(item);
+    }
+
+    private void loadItems() {
+        Connection connection;
+        try {
+            connection = ConnectionFactory.getUsableConnection();
+        } catch (ConnectionException e) {
+            connection = null;
+        }
+        if (connection == null) {
+            updateViewVisibility(false, true);
+            return;
+        }
+
+        mItemPickerAdapter.clear();
+        updateViewVisibility(true, false);
+
+        final AsyncHttpClient client = connection.getAsyncHttpClient();
+        mRequestHandle = client.get("rest/items", new AsyncHttpClient.StringResponseHandler() {
+            @Override
+            public void onSuccess(String responseBody, Headers headers) {
+                try {
+                    ArrayList<Item> items = new ArrayList<>();
+                    JSONArray jsonArray = new JSONArray(responseBody);
+                    for (int i = 0; i < jsonArray.length(); i++) {
+                        JSONObject itemJson = jsonArray.getJSONObject(i);
+                        Item item = Item.fromJson(itemJson);
+                        if (!item.readOnly()) {
+                            items.add(item);
+                        }
+                    }
+                    Log.d(TAG, "Item request success, got " + items.size() + " items");
+                    mItemPickerAdapter.addLoadedItems(items);
+                    handleInitialHighlight();
+                    updateViewVisibility(false, false);
+                } catch (JSONException e) {
+                    Log.d(TAG, "Item response could not be parsed", e);
+                    updateViewVisibility(false, true);
+                }
+            }
+
+            @Override
+            public void onFailure(Request request, int statusCode, Throwable error) {
+                updateViewVisibility(false, true);
+                Log.e(TAG, "Item request failure", error);
+            }
+        });
+    }
+
+    @Override
+    public void onClick(View view) {
+        if (view == mRetryButton) {
+            loadItems();
+        }
+    }
+
+    @Override
+    public void onItemClicked(Item item) {
+        if (item == null) {
+            return;
+        }
+
+        ArrayList<String> labels = new ArrayList<>();
+        ArrayList<String> commands = new ArrayList<>();
+
+        if (item.isOfTypeOrGroupType(Item.Type.Switch)) {
```

Comment:
This is basically the same mapping (item to possible/common) state as in WidgetListFragment, except that WidgetListFragment also looks at the widget.
@maniac103 Do you think it makes sense to add the functions `addCommandsForItem(Item item, ArrayList<String> labels,ArrayList<String> commands)` and `addCommandsForWidget(Widget widget, ArrayList<String> labels,ArrayList<String> commands)`, use the forItem function here. And to use both in WidgetListFragment and remove duplicate commands from the ArrayLists?